### PR TITLE
Remove Optional

### LINKED ISSUE #11: Remove unnecessary null checks by leveraging kotlin non-nullable types
Some classes (like RpcMapper) are checking if some non-null types are null, which should not be necessary. We should remove those checks and better leverage kotlin's null safety features across the project.

### PR DESCRIPTION
**Pull request by dddj698**
Remove Optional as Kotlin is null-safe

Fix #11

### CODE REVIEW

**Comment by dddj698** [comment]
Need to rebase after PR21

File: src/main/kotlin/org/eclipse/uprotocol/cloudevent/validate/CloudEventValidator.kt [hunk 1]
```diff
@@ -114,12 +113,11 @@ abstract class CloudEventValidator {
         }
 
         override fun validateSink(cloudEvent: CloudEvent): ValidationResult {
-            val maybeSink: Optional<String> = UCloudEvent.getSink(cloudEvent)
-            if (maybeSink.isPresent) {
-                val sink: String = maybeSink.get()
+            var result: ValidationResult = ValidationResult.success()
```
Comment (mikaelperrin) [inline src/main/kotlin/org/eclipse/uprotocol/cloudevent/validate/CloudEventValidator.kt:116]: Could we remove the mutable state `result` here or would that become unreadable?
Maybe multiple returns is better in this situation (similar to `Notification::validateSink`)?

File: src/main/kotlin/org/eclipse/uprotocol/cloudevent/validate/CloudEventValidator.kt [hunk 2]
```diff
@@ -241,21 +231,18 @@ abstract class CloudEventValidator {
...
-                        )
+            val sink: String = UCloudEvent.getSink(cloudEvent)
+                ?: return ValidationResult.failure("Invalid CloudEvent sink. Response CloudEvent sink must be uri the destination of the response.")
+            val checkSink: ValidationResult = validateRpcTopicUri(sink)
+            if (checkSink.isFailure()) {
+                return ValidationResult.failure(
+                    String.format(
+                        "Invalid RPC Response CloudEvent sink [%s]. %s",
+                        sink,
+                        checkSink.getMessage()
```
Comment (mikaelperrin) [inline src/main/kotlin/org/eclipse/uprotocol/cloudevent/validate/CloudEventValidator.kt:242]: Sounds like all those validateSink are very similar. Should we refactor with a private method here?

File: src/main/kotlin/org/eclipse/uprotocol/uri/serializer/MicroUriSerializer.kt [hunk 3]
```diff
@@ -59,12 +60,12 @@ class MicroUriSerializer private constructor() : UriSerializer<ByteArray> {
      * @param uri The [UUri] data object.
      * @return Returns a byte[] representing the serialized [UUri].
      */
-    override fun serialize(uri: UUri?): ByteArray {
-        if (uri == null || uri.isEmpty() || !uri.isMicroForm()) {
+    override fun serialize(uri: UUri): ByteArray {
+        if (uri.isEmpty() || !uri.isMicroForm()) {
```
Comment (mikaelperrin) [inline src/main/kotlin/org/eclipse/uprotocol/uri/serializer/MicroUriSerializer.kt:64]: Is this condition right? if the uri is not micro form, we serialize with an empty array? It's not related to your patch but that seems odd.

Comment (dddj698) [reply src/main/kotlin/org/eclipse/uprotocol/uri/serializer/MicroUriSerializer.kt:64]: it's from MicroUriSerializer, I think it's reasonable to serialize with an empty array if it is not in micro form.

File: src/main/kotlin/org/eclipse/uprotocol/transport/UPayloadUtils.kt
Comment (mikaelperrin) [inline src/main/kotlin/org/eclipse/uprotocol/transport/UPayloadUtils.kt:1]: Out of consistency and kotlin naming convention, should we name this one UPayloadExt? 
Also, do we really need both the "util" method and the extension function?

**Review by mikaelperrin — APPROVED** [review]

**Pull request merged**